pkg/asset/installconfig/ibmcloud/metadata.go

```diff
@@ -92,6 +100,50 @@ func (m *Metadata) SetCISInstanceCRN(crn string) {
 	m.cisInstanceCRN = crn
 }
 
+// DNSInstance returns a DNSInstance holding information about the DNS Services instance
+// managing the DNS zone for the base domain.
+func (m *Metadata) DNSInstance(ctx context.Context) (*DNSInstance, error) {
+	if m.dnsInstance != nil {
+		return m.dnsInstance, nil
+	}
```

Comment:
Do we want to allow access of this variable outside the protection of the mutex?

Reply:
I think that is acceptable, since if the value isn't set, it get locked by the mutex after this check, setting it, or waiting for another thread to set it and then rechecking if it is set.

I thought this method was potentially best given the last time I ran into this when fixing the deadlock bug, per this review comment
https://github.com/openshift/installer/pull/6241#pullrequestreview-1077917887

Not opposed to making a change, but just wasn't sure which path is safer/best.